docs/send-data/opentelemetry-collector/data-transformations.md

```diff
@@ -0,0 +1,112 @@
+---
+id: data-transformations
+title: OpenTelemetry - String Hashing and Masking using Transform Processor and OTTL
+sidebar_label: String Hashing and Masking
+description: Learn how to perform string hashing and masking operations using the Transform Processor and OTTL in OpenTelemetry.
+---
+
+OpenTelemetry provides the Transform Processor and OTTL (OpenTelemetry Transformation Language), empowering you to perform string hashing and masking operations on telemetry data. With the flexibility to configure the Transform Processor in your OpenTelemetry pipeline, you can replace sensitive information with hashed values or masked strings, ensuring data protection.
+
+You can find more detailed information about the available OTTL functions and their usage in the [OTTL Functions README](https://github.com/rnishtala-sumo/opentelemetry-collector-contrib/blob/ottl-replace-pattern/pkg/ottl/ottlfuncs/README.md).
+
+## Hashing examples
+
+### Example 1: Hashing a password in log body
+
+```yaml
+processors:
+  attributes/extract:
+  actions:
+    - key: message
+      pattern: "^password=(?P<password>\\w+)$"
+      action: extract
+
+  transform/replace:
+    log_statements:
+      - context: log
+        statements:
+          - set(attributes["password"], SHA256(attributes["password"]))
+          - set(attributes["password"], Concat(["passwd", attributes["password"]], "="))
+          - replace_pattern(attributes["message"], "password=(test)", attributes["password"])
+```
+
+The configuration consists of two sections: `attributes/extract` and `transform/replace`, representing attribute extraction and transformation operations, respectively.
+
+- In the `attributes/extract` section, the `actions` key specifies the extraction action. In this example, the action extracts the password from an attribute value using a regular expression pattern. The pattern `^password=(?P<password>\\w+)$` captures the password value after `password=` and assigns it to the `password` attribute.
+
+- The `transform/replace` section defines a transformation operation using the Transform Processor. Within the `log_statements` context, the following statements are executed:
+
+  - The `set` function is used to hash the `password` attribute using the SHA256 hash algorithm. The resulting hash value replaces the original `password` value.
+
+  - The `set` function is used to concatenate the string prefix `"hashed"` with the hashed `password`. This creates a new `password` value with the prefix included.
+
+  - The `replace_pattern` function is applied to the `message` attribute. If the value matches the pattern `"password=(test)"`, the matched section is replaced with the hashed `password` attribute.
+
+### Example 2: Hashing a Kubernetes name
```

Comment:
Am I missing some material difference between example 1 and 2? Seems like a lot of duplicated content just to communicate that this hashing pattern can be applied to more than passwords.

Reply:
I wanted to illustrate two different examples. I could change the hash function in the second one to add more value.

Reply:
Makes sense!

Maybe hashing an attribute - just so that nobody gets confused about what hashing's got to do with the whole extract/hash/concat/replace contraption?

Reply:
made this change.